Mathlib/FieldTheory/PrimitiveElement.lean

```diff
@@ -282,75 +273,49 @@ private lemma finite_intermediateField_of_exists_primitive_element.aux_2
-    (mem_inf (S := K) (T := K')).2
-      ⟨by rw [coeff_map]; exact ((minpoly K α).coeff n).property,
-        by rw [heq, coeff_map]; exact ((minpoly K' α).coeff n).property⟩⟩ * X ^ n
-  have hp : p.map (algebraMap K'' E) = g := by
-    rw [← sum_C_mul_X_pow_eq g, Polynomial.sum_def, Polynomial.map_sum]
-    congr
-    funext n
-    rw [Polynomial.map_mul, Polynomial.map_pow, Polynomial.map_X, Polynomial.map_C]
-    congr
-  let q := minpoly K'' α
-  obtain ⟨r, hqdvdp⟩ := show q ∣ p by
-    apply minpoly.dvd
-    rw [aeval_def, eval₂_eq_eval_map, hp, ← eval₂_eq_eval_map, ← aeval_def]
+  have g_lifts : g ∈ lifts (algebraMap ↑(K ⊓ K') E)
+  · refine g.lifts_iff_coeff_lifts.mpr fun n ↦ ?_
+    erw [Subtype.range_val]; constructor
+    · rw [coeff_map]; apply Subtype.mem
+    · rw [heq, coeff_map]; apply Subtype.mem
+  obtain ⟨p, hp⟩ := g.lifts_and_natDegree_eq_and_monic
+    g_lifts ((minpoly.monic <| isIntegral_of_finite K α).map _)
+  have dvd_p : minpoly ↑(K ⊓ K') α ∣ p
+  · apply minpoly.dvd
+    rw [aeval_def, eval₂_eq_eval_map, hp.1, ← eval₂_eq_eval_map, ← aeval_def]
     exact minpoly.aeval K α
-  have hpne0 : q * r ≠ 0 := by
-    apply_fun Polynomial.map (algebraMap K'' E)
-    rw [← hqdvdp, hp, Polynomial.map_zero,
-      Polynomial.map_ne_zero_iff $ NoZeroSMulDivisors.algebraMap_injective K E]
-    exact minpoly.ne_zero $ isAlgebraic_iff_isIntegral.1 $ isAlgebraic_of_finite K α
-  have hpdeg := IntermediateField.adjoin.finrank (K := K) (x := α) (hx := isIntegral_of_finite K α)
-  letI : Algebra K (⊤ : IntermediateField K E) := IntermediateField.algebra _
-  letI : Module K (⊤ : IntermediateField K E) := Algebra.toModule
-  letI : Algebra K'' (⊤ : IntermediateField K'' E) := IntermediateField.algebra _
-  letI : Module K'' (⊤ : IntermediateField K'' E) := Algebra.toModule
-  rw [IntermediateField.adjoin_eq_top_of_adjoin_eq_top F K E hprim,
-    show finrank K (⊤ : IntermediateField K E) = finrank K E from finrank_top K E,
-    ← natDegree_map_eq_of_injective (NoZeroSMulDivisors.algebraMap_injective K E) (minpoly K α),
-    show (minpoly K α).map (algebraMap K E) = g from rfl,
-    ← hp,
-    natDegree_map_eq_of_injective (NoZeroSMulDivisors.algebraMap_injective K'' E),
-    hqdvdp,
-    natDegree_mul (left_ne_zero_of_mul hpne0) (right_ne_zero_of_mul hpne0),
-    ← IntermediateField.adjoin.finrank (K := K'') (x := α) (hx := isIntegral_of_finite K'' α),
-    IntermediateField.adjoin_eq_top_of_adjoin_eq_top F K'' E hprim,
-    show finrank K'' (⊤ : IntermediateField K'' E) = finrank K'' E from finrank_top K'' E] at hpdeg
-  replace hpdeg : finrank K E ≥ finrank K'' E := by linarith only [hpdeg, Nat.le_add_right]
-  exact inf_eq_left.1 (eq_of_le_of_finrank_le' inf_le_left hpdeg)
+  have finrank_eq : ∀ K : IntermediateField F E, finrank K E = natDegree (minpoly K α)
+  · intro K
+    have := adjoin.finrank (isIntegral_of_finite K α)
+    erw [adjoin_eq_top_of_adjoin_eq_top F K E hprim, finrank_top K E] at this
```

Comment:
Quite weird (and fortunate) that `rw [adjoin_eq_top_of_adjoin_eq_top F K E hprim]` doesn't result in a "motive not type correct error" and requires introduction of no instance.